src/Dsll.hs

```diff
@@ -148,13 +148,21 @@ cd1 = do
     colon
     t' <- typeParser
     pos <- getPosition
-    return Cd { nameCd = name, inputCd = zip y k, outputCd = t' }
+    return (CdStmt Cd { nameCd = name, inputCd = zip y k, outputCd = t' })
 cd2 = do
     rword "tconstructor"
     name <- identifier
     colon
     t' <- typeParser
-    return Cd { nameCd = name, inputCd = [], outputCd = t' }
+    return (CdStmt Cd { nameCd = name, inputCd = [], outputCd = t' })
+
+-- | sub type declarations parser
+stdParser :: Parser DsllStmt
+stdParser = do
+  subtype <- tParser
+  rword "<:"
+  supertype <- tParser
+  return (StdStmt (Std { subType = subtype, superType = supertype}))
```

Comment:
you can also write `return $ StdStmt $ Std { subType = subtype, superType = supertype}`. often application is easier to read and write than a lot of parens